### ISSUE TITLE
Support specifying a typed  schema for db

### ISSUE THREAD

**Issue by R-omk** [vitaliy-art/tarantoolscript#34]
TypeScript offers amazing possibilities, one of which is "type inference."

The project most similar in spirit from the world of JavaScript is IndexedDB. In the example from the link below, you can see how a declared type allows you to both create tables with the required schema and later use that type when performing operations with data in the database.

https://github.com/jakearchibald/idb?tab=readme-ov-file#typescript

**Comment by vitaliy-art** [comment]
Yeah, i'm planning add something like typed generics in retrieval methods. Something like `box.space.get<Band>('bands').get(key)` or, maybe, `box.space.get('bands').get<Band>(key)`. I'm sure it'll be so usefull feature
